### ISSUE TITLE
[java] Documentation of ignoredAnnotations property is misleading

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#1057]
**Affects PMD Version:** 
6.2.0

**Rule:**
`ImmutableField`, `SingularField`, `UnusedPrivateField` and `UnusedPrivateMethod` 

**Description:**
[Documentation](https://pmd.github.io/pmd-6.2.0/pmd_rules_java_bestpractices.html#unusedprivatefield) suggests that the format is `
[lombok.Setter, lombok.Getter, lombok.Builder, lombok.Data, lombok.RequiredArgsConstructor, lombok.AllArgsConstructor, lombok.Value, lombok.NoArgsConstructor, java.lang.Deprecated, javafx.fxml.FXML]`. But the implementation supports only pipe-separated format without brackets at the beginning and at the end.

**Code Sample demonstrating the issue:**

```xml
    <rule ref="category/java/bestpractices.xml/UnusedPrivateField">
        <properties>
            <property name="ignoredAnnotations" value="[java.lang.Deprecated, com.thoughtworks.xstream.annotations.XStreamAlias]"/>
        </properties>
    </rule>
```
The above config doesn't work and doesn't produce any warning about incorrect usage.

Correct configs seems to be following:

```xml
    <rule ref="category/java/bestpractices.xml/UnusedPrivateField">
        <properties>
            <property name="ignoredAnnotations" value="java.lang.Deprecated|com.thoughtworks.xstream.annotations.XStreamAlias"/>
        </properties>
    </rule>
```
**Running PMD through:** Gradle

**Comment by adangel** [comment]
Thanks for reporting! It looks like, we just output the List of default values as string, but we should use the same format, that is expected when using the property.
